Doc/.ruff.toml

```diff
@@ -0,0 +1,55 @@
+target-version = "py312"  # align with the version in oldest_supported_sphinx
+fix = true
+output-format = "full"
+extend-exclude = [
+    "includes/*",
+    "tools/extensions/c_annotations.py",
+    "tools/extensions/escape4chm.py",
+    "tools/extensions/patchlevel.py",
+    "tools/extensions/pyspecific.py",
+]
+
+[lint]
+preview = true
+select = [
+    "C4",  # flake8-comprehensions
+    "B",   # flake8-bugbear
+    "E",   # pycodestyle
+    "EM",  # flake8-errmsg
```

Comment:
Some of the `flake8-errmsg` ones are bit too opinionated for my taste (though I don't feel too strongly)

Reply:
To the bin!